routes/profile.js

```diff
@@ -1,136 +1,118 @@
 const express = require('express');
-const _ = require('lodash');
 
 const Selection = require('../models/selection');
-const buildProfileSQL = require('../query-helpers/profile');
-const buildProfileSingleSQL = require('../query-helpers/profile-single');
-const buildDecennialSQL = require('../query-helpers/decennial');
-const tableConfigs = require('../table-config');
-const delegateAggregator = require('../utils/delegate-aggregator');
-const nestProfile = require('../utils/nest-profile');
+const profileQuery = require('../query-helpers/profile');
+const decennialQuery = require('../query-helpers/decennial');
+const DataIngestor = require('../utils/data-ingestor');
+const doChangeCalculations = require('../utils/change');
+const doDifferenceCalculations = require('../utils/difference');
 
 const router = express.Router();
 
+router.get('/:id/:profile', async (req, res) => {
+  const { app } = req;
+  const { id: _id, profile } = req.params;
+  const { compare = '0' } = req.query;
+
+  if (invalidCompare(compare)) res.status(500).send({ error: 'invalid compare param' });
+
+  if (invalidProfile(profile)) res.status(500).send({ error: 'invalid profile' });
 
-router.use((req, res, next) => {
-  res.setHeader('Cache-Control', 'public, max-age=0');
-  next();
+  try {
+    const selectedGeo = await Selection.findOne({ _id });
+    const profileObj = await getProfileData(profile, selectedGeo.geoids, compare, app.db);
```

Comment:
This is great and reads clearly... although I was a little surprised that `getProfileData` is not imported but instead defined later in this route file. I know this was prevented by a linting rule, and I have wanted to do this for smaller utility functions (things like dasherizing or do some date parsing), but `getProfileData` seems a bit more involved. I'm not quite sure what I'm getting back, and it is after all doing all the work... (maybe this is why it _should_ live here and not tucked away as a "utility"? 🦆) Thoughts?

Reply:
Separate: I know this was how we named things before the refactor, but I'm wondering if there might be better naming to distinguish between the `profile` query parameter (provided by the route params) and the `profileObj` which is the fully generated data. Maybe we could alias it to something like `profileName`? (I realize naming is low-hanging fruit, just sorta reacting here - if it's helpful - after having not looked at any of this code in many many months :) )

Reply:
naming is not low hanging fruit if you ask me! I definitely think `profile` qp should become `profileName`

also, happy to extract the `getProfileData` into a separate place. I extracted it b/c i was thinking it could be repurposed in a route that does not require a selection ID but rather accepts a list of geoids -- wouldn't be used in the front end, but could be useful to people like that grad student that emailed & wanted to scrape a bunch of profile data; they could programatically hit that route, esp if we allow 'compare' to be optional. but I agree that I don't necessarily love the idea of this living in a `util`. Let's think about where else this could go ?